src/lib/core/components/computations/plugins/abundance.tsx

```diff
@@ -153,6 +155,23 @@ export function AbundanceConfiguration(props: ComputationConfigProps) {
     visualizationId
   );
 
+  const selectedDataConfig = collections.find(
+    (collectionVar) =>
+      collectionVar.id === collectionVariable.variableId &&
+      collectionVar.entityId === collectionVariable.entityId
+  );
+  const dataConfigButtonDisplay =
+    selectedDataConfig?.entityDisplayName +
+    ' > ' +
+    selectedDataConfig?.displayName;
+
+  const [selectedDataValue, setSelectedDataValue] = useState('');
+
+  useEffect(() => {
+    if (!selectedDataValue.length) return;
+    changeConfigHandler('collectionVariable', JSON.parse(selectedDataValue));
+  }, [selectedDataValue]);
```

Comment:
The `useState` and `useEffect` hooks seem unnecessary here. It looks like you are doing this because the value needs `JSON.parse()` to be applied to it, and the combination of this `useState` and `useEffect` hook allows you to use the `changeConfigHandler` function, while also using it for changing ranking method.

I think a preferable solution would be to do something like this:
```
<SingleSelect
  ...
  onSelect={flow(
    JSON.parse,
    partial(changeConfigHandler, 'collectionVariable')
  )}
  ...
/>
```
This follows the same pattern you're using for rankingMethod.

An even better solution would be to allow `SingleSelect` values to be anything, rather than just strings. Then, you wouldn't need to using `JSON.stringify` and `JSON.parse`. I can make an issue for this in coreui.

This all applies to the `alphaDiv` plugin, as well.